suzieq/poller/services/interfaces.py

```diff
@@ -428,17 +436,18 @@ def fix_junos_speed(entry):
                 if not macaddr:
                     macaddr = entry.get('macaddr', '00:00:00:00:00:00')
 
+                addrlist = []
                 for elem in afis:
                     afi_mtu = elem.get('mtu', [{}])[0].get(
                         'data', entry['mtu'])
                     afi = elem.get('address-family-name', [{}])[0] \
                         .get('data', '')
                     if afi == 'inet':
                         no_inet = False
-                        addrlist = elem.get('interface-address', [])
+                        addrlist += elem.get('interface-address', [])
```

Comment:
Is `elem['interface-address']` always an array? If not it could crash here

Reply:
yes, always an array. See the init above

Reply:
That is `addrList`, I meant `elem['interface-address']` from the `processed_data`